src/libraries/System.Text.RegularExpressions/src/System/Text/RegularExpressions/RegexRunner.cs

```diff
@@ -57,6 +59,10 @@ public abstract class RegexRunner
         protected internal Match? runmatch;        // result object
         protected internal Regex? runregex;        // regex object
 
+        internal int originalRuntextbeg;           // In the CompiledToAssembly case, it is important to store the original runtexbeg
+                                                   // that was passed in from the user, mainly because it works over the original input
+                                                   // as opposed to working over the sliced span.
```

Comment:
Hmm.  While minor, it'd be really nice not to add overhead like additional fields to all uses purely in support of a legacy scenario we don't really care about other than ensuring it doesn't blow up.

Could we instead compute this in Scan below with https://docs.microsoft.com/en-us/dotnet/api/system.memoryextensions.overlaps?view=net-6.0#system-memoryextensions-overlaps-1(system-readonlyspan((-0))-system-readonlyspan((-0))-system-int32@) between the span and the string?

Reply:
I wasn't familiar with Overlaps method, I'll try it out and see if this is possible. I did originally wanted to calculate this based on the difference in lengths of span and string, but as I pointed out the issue was that we couldn't guarantee that beginning = string.Length - span.Length